packages/tools/nodeEditor/src/graphSystem/connectionPointPortData.ts

```diff
@@ -163,8 +163,14 @@ export class ConnectionPointPortData implements IPortData {
         }
 
         switch (issue) {
-            case NodeMaterialConnectionPointCompatibilityStates.TypeIncompatible:
-                return `Cannot connect two different connection types:\nSource is ${NodeMaterialBlockConnectionPointTypes[this.data.type]} and destination is ${NodeMaterialBlockConnectionPointTypes[(targetPort.data as NodeMaterialConnectionPoint).type]}`;
+            case NodeMaterialConnectionPointCompatibilityStates.TypeIncompatible: {
+                const port = targetPort.data as NodeMaterialConnectionPoint;
+                let acceptedTypes = port.acceptedConnectionPointTypes.map((t) => NodeMaterialBlockConnectionPointTypes[t]).join(", ");
+
+                acceptedTypes = `${NodeMaterialBlockConnectionPointTypes[port.type]}` + (acceptedTypes ? `,${acceptedTypes}` : "");

```

Comment:
nit:

```suggestion
                acceptedTypes = `${NodeMaterialBlockConnectionPointTypes[port.type]}` + (acceptedTypes ? `, ${acceptedTypes}` : "");
```

Same below.